kabinizer-back-end/kabinizer-api/Services/Draw/DrawService.cs

```
@@ -24,10 +20,86 @@ public void CreateDraw(CreateDrawDto draw)
             DeadlineStart = draw.DeadlineStart,
             DeadlineEnd = draw.DeadlineEnd,
             Title = draw.Title,
-            Periods = periodService.CreatePeriods(drawId, draw.IsSpecial, draw.DrawPeriods),
+            Periods = periodService.CreatePeriods(drawId, draw.DrawPeriods),
             IsSpecial = draw.IsSpecial
         };
         entityContext.Draws.Add(drawEntity);
-        entityContext.SaveChanges();
+        await entityContext.SaveChangesAsync();
+        return drawEntity;
+    }
+
+    /**
+     * Get a draw
+     */
+    public async Task<DrawEntity> GetDraw(Guid drawId)
+    {
+        DrawEntity drawEntity = await entityContext.Draws.FindAsync(drawId) ?? throw new Exception("Draw not found");
+        return drawEntity;
+    }
+
+    /**
+     * Update a draw
+     */
+    public Task UpdateDraw(Guid drawId)
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * Delete a draw
+     */
+    public async Task DeleteDraw(Guid drawId)
+    {
+        DrawEntity drawEntity = await entityContext.Draws.FindAsync(drawId) ?? throw new Exception("Draw not found");
+        entityContext.Draws.Remove(drawEntity);
+        await entityContext.SaveChangesAsync();
+    }
```

`   public void DeleteDraw(string id)
    {
        var draw = entityContext.Draws.Find(Guid.Parse(id)) ?? throw new Exception("Draw not found");
        entityContext.Draws.Remove(draw);
        entityContext.SaveChanges();
    }`

min kode for DeleteDraw 🥇